code/game/turfs/simulated/floor.dm

```diff
@@ -30,6 +30,7 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 	var/list/broken_states = list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
 	var/list/burnt_states = list("floorscorched1", "floorscorched2")
 	var/list/prying_tool_list = list(TOOL_CROWBAR) //What tool/s can we use to pry up the tile?
+	var/keep_dir = TRUE
```

Comment:
This is very interesting. What's the purpose of this addition?

Might be a good idea to make the variable's name describe *when* the dir is kept.

Reply:
When the glass floor inherits the dir of the plating turf, anything other than 2 screws with the smoothing in a way I'm unsure is fixable without redoing bits of smoothing code. Turfs with this var inherit the dir of the previous turf on changeturf()